optimism/src/mips/witness.rs

```diff
@@ -808,13 +829,19 @@ impl<Fp: Field> Env<Fp> {
         self.scratch_state = fresh_scratch_state();
     }
 
-    pub fn write_column(&mut self, column: Column, value: u64) {
-        self.write_field_column(column, value.into())
+    pub fn reset_inverse_state(&mut self) {
+        self.inverse_state_idx = 0;
+        self.inverse_state = fresh_inverse_state();
     }
 
-    pub fn write_field_column(&mut self, column: Column, value: Fp) {
+    pub fn write_column(&mut self, column: Column, value: u64) {
         match column {
             Column::ScratchState(idx) => self.scratch_state[idx] = value,
+            Column::InverseState(idx) => {
+                // So that we won't try to invert the zero value
+                assert_ne!(value, 0);
```

Comment:
```
FILENAME=2023-12-13-15-02-00-op-program-data-log.sh bash run-code.sh
```
raises an error here.

Reply:
That's because I am allocating inverse state even when sometimes x is zero, will have to fix this. thanks!